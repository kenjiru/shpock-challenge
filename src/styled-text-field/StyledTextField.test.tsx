import * as React from "react";
import { mount, shallow, ReactWrapper } from "enzyme";
import shallowToJson from "enzyme-to-json";
import { TextField, TextFieldProps } from "material-ui";
import { MuiThemeProvider } from "material-ui/styles";

import ThemedStyledTextField, { IStyledTextFieldProps, StyledTextField } from "./StyledTextField";

describe("StyledTextField", () => {
    const ID: string = "foo";

    it("should render correctly", () => {
        const output = shallow(
            <StyledTextField id={ID} hasError={false}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it("should have the correct styles when hasError is true", () => {
        const output: ReactWrapper = mount(
            <MuiThemeProvider>
                <ThemedStyledTextField id={ID} hasError={true}/>
            </MuiThemeProvider>
        );

        const styledTextFieldProps: IStyledTextFieldProps = output.find(StyledTextField).props();
        const textFieldProps: TextFieldProps = output.find(TextField).props();

        expect(textFieldProps.underlineStyle.borderColor).toBe(styledTextFieldProps.muiTheme.textField.errorColor);
        expect(textFieldProps.underlineFocusStyle.borderColor).toBe(styledTextFieldProps.muiTheme.textField.errorColor);

        expect(textFieldProps.inputStyle.color).toBe(styledTextFieldProps.muiTheme.textField.errorColor);
        expect(textFieldProps.floatingLabelStyle.color).toBe(styledTextFieldProps.muiTheme.textField.errorColor);
    });

    it("should have the correct styles when hasError is false", () => {
        const output: ReactWrapper = mount(
            <MuiThemeProvider>
                <ThemedStyledTextField id={ID} hasError={false}/>
            </MuiThemeProvider>
        );

        const textFieldProps: TextFieldProps = output.find(TextField).props();

        expect(textFieldProps.underlineStyle.borderColor).toBeUndefined();
        expect(textFieldProps.underlineFocusStyle.borderColor).toBeUndefined();

        expect(textFieldProps.inputStyle.color).toBeUndefined();
        expect(textFieldProps.floatingLabelStyle.color).toBeUndefined();
    });
});
