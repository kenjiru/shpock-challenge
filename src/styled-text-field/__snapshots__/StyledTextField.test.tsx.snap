// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledTextField should render correctly 1`] = `
<TextField
  disabled={false}
  floatingLabelFixed={false}
  floatingLabelStyle={Object {}}
  fullWidth={false}
  id="foo"
  inputStyle={Object {}}
  multiLine={false}
  rows={1}
  type="text"
  underlineFocusStyle={Object {}}
  underlineShow={true}
  underlineStyle={Object {}}
/>
`;
